app/src/main/java/com/example/playlistmaker/Creator.kt

```diff
@@ -0,0 +1,37 @@
+package com.example.playlistmaker
+
+import android.content.Context
+import com.example.playlistmaker.data.PlayerImpl
+import com.example.playlistmaker.data.storage.SearchHistoryStorageImpl
+import com.example.playlistmaker.domain.api.AudioPlayerInteractor
+import com.example.playlistmaker.domain.api.Player
+import com.example.playlistmaker.domain.api.SearchHistoryStorage
+import com.example.playlistmaker.domain.impl.AudioPlayerInteractorImpl
+import com.example.playlistmaker.presentation.api.AudioPlayer
+import com.example.playlistmaker.presentation.api.AudioPlayerPresenter
+import com.example.playlistmaker.presentation.presenters.AudioPlayerPresenterImpl
+
+object Creator {
+
+    fun createAudioPlayerPresenter(context: Context, audioPlayer: AudioPlayer): AudioPlayerPresenter {
+        return AudioPlayerPresenterImpl(
+            audioPlayer = audioPlayer,
+            audioPlayerInteractor = createAudioPlayerInteractor(context)
+        )
+    }
+
+    private fun createAudioPlayerInteractor(context: Context): AudioPlayerInteractor {
+        return AudioPlayerInteractorImpl(
+            player = createPlayer(),
+            searchHistoryStorage = createSearchHistoryStorage(context)
+        )
+    }
+
+    private fun createPlayer(): Player {
+        return PlayerImpl()
+    }
```

Comment:
```suggestion
    private fun createPlayer() = PlayerImpl()
```

Это не замечание, а рекомендация